flareon-cli/src/migration_generator.rs

```diff
@@ -84,44 +109,76 @@ impl MigrationGenerator {
             .canonicalize()
             .with_context(|| "unable to canonicalize src dir")?;
 
-        let mut source_files = Vec::new();
+        let source_file_paths = Self::find_source_files(&src_dir)?;
+        let source_files = source_file_paths
+            .into_iter()
+            .map(|path| {
+                Self::parse_file(&src_dir, path.clone())
+                    .with_context(|| format!("unable to parse file: {path:?}"))
+            })
+            .collect::<anyhow::Result<Vec<_>>>()?;
+        Ok(source_files)
+    }
+
+    fn find_source_files(src_dir: &Path) -> anyhow::Result<Vec<PathBuf>> {
+        let mut paths = Vec::new();
         for entry in glob::glob(src_dir.join("**/*.rs").to_str().unwrap())
             .with_context(|| "unable to find Rust source files with glob")?
         {
             let path = entry?;
-            source_files.push(path);
+            paths.push(
+                path.strip_prefix(src_dir)
+                    .expect("path must be in src dir")
+                    .to_path_buf(),
+            );
         }
 
-        Ok(source_files)
+        Ok(paths)
     }
 
-    fn process_source_files(&self, paths: &Vec<PathBuf>) -> anyhow::Result<AppState> {
+    fn process_source_files(&self, paths: Vec<SourceFile>) -> anyhow::Result<AppState> {
```

Comment:
If the function has been modified to accept `SourceFile` instead of `PathBuf`, just for consistency sake, it would make sense to rename `paths` to `files` or similar.

Reply:
Right, missed this during a refactor. Fixed!